src/go/rdctl/cmd/snapshot.go

```diff
@@ -43,20 +44,35 @@ func exitWithJsonOrErrorCondition(e error) error {
 		snapshotErrors = append(snapshotErrors, e)
 	}
 	if outputJsonFormat {
+		exitStatus := 0
 		for _, snapshotError := range snapshotErrors {
 			if snapshotError != nil {
+				exitStatus = 1
 				errorPayload.Error = snapshotError.Error()
 				jsonBuffer, err := json.Marshal(errorPayload)
 				if err != nil {
-					return fmt.Errorf("error json-converting error messages: %w", err)
+					snapshotErrors = append(snapshotErrors, fmt.Errorf("error json-converting error messages: %w", err))
+					return errors.Join(snapshotErrors...)
 				}
 				fmt.Fprintf(os.Stdout, string(jsonBuffer)+"\n")
 			}
 		}
-		return nil
-	} else {
-		return errors.Join(snapshotErrors...)
+		os.Exit(exitStatus)
+	}
+	return errors.Join(snapshotErrors...)
+}
+
+func getSnapshotId(manager snapshot.Manager, idOrName string) (string, error) {
+	snapshots, err := manager.List()
+	if err != nil {
+		return "", fmt.Errorf("failed to list snapshots: %w", err)
+	}
+	for _, candidate := range snapshots {
+		if idOrName == candidate.Name || idOrName == candidate.ID {
```

Comment:
Would it be possible to always prefer id over name (if a name exactly matches the id of a different snapshot)?

Reply:
I overlooked that. I would prefer if we put IDs and names in one namespace, so you can't name a snapshot with an existing ID, and the system won't generate an ID that matches the name of an existing snapshot.

Reply:
I would prefer if we don't expose IDs at all; they are an implementation detail. We already enforce that names are unique, so names are the only thing the user should ever see.

The only reason we have IDs at all is because a name may include characters that are not valid in a filename (like `:` on NTFS). Otherwise we could have used the name as the directory name.

So please drop deleting by ID completely. The name is the unique key to identify a snapshot.

Reply:
So this means:
1. `snapshot list` no longer shows the ID field, for both table and json views
2. `snapshot remove` takes a snapshot name
3. `snapshot restore` takes a snapshot name
4. You can create a snapshot with a name that happens to match another existing snapshot's ID